tests/integration/tasks/helpers/requestSender.ts

```diff
@@ -0,0 +1,9 @@
+import * as supertest from 'supertest';
+
+export class TasksRequestSender {
+  public constructor(private readonly app: Express.Application) {}
+
+  public async notifyFinished(taskId: string): Promise<supertest.Response> {
```

Comment:
I changed the function name to `handleTaskNotification` in the taskController. Specifically in the supertest requestSender, I don't think that it really fits because it doesn't handle the notification, but rather sends the notification. I did change it to `notifyTaskFinished` as I think it's clearer and more readable.